packages/jest-worker/src/base/BaseWorkerPool.js

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+'use strict';
+
+import mergeStream from 'merge-stream';
+import os from 'os';
+import path from 'path';
+
+import ChildProcessWorker from '../workers/ChildProcessWorker';
+import NodeThreadsWorker from '../workers/NodeThreadsWorker';
+import {CHILD_MESSAGE_END} from '../types';
+
+import type {Readable} from 'stream';
+import type {FarmOptions, WorkerOptions, WorkerInterface} from '../types';
+
+/* istanbul ignore next */
+const emptyMethod = () => {};
+
+export default class BaseWorkerPool {
+  _stderr: Readable;
+  _stdout: Readable;
+  _options: FarmOptions;
+  _workers: Array<WorkerInterface>;
+
+  constructor(workerPath: string, options: FarmOptions) {
+    this._options = options;
+
+    const numWorkers = options.numWorkers || os.cpus().length - 1;
+    this._workers = new Array(numWorkers);
+
+    if (!path.isAbsolute(workerPath)) {
+      workerPath = require.resolve(workerPath);
+    }
+
+    const stdout = mergeStream();
+    const stderr = mergeStream();
+
+    for (let i = 0; i < numWorkers; i++) {
+      const workerOptions: WorkerOptions = {
+        forkOptions: options.forkOptions || {},
+        maxRetries: options.maxRetries || 3,
+        useNodeWorkersIfPossible: options.useNodeWorkersIfPossible,
```

Comment:
Let's remove this and make it `true` always.

Reply:
It makes sense as an option, doesn't it? Seems like an opt-in thing (beyond using a version of node where it's unflagged)

Reply:
There is no real reason why you'd like to use `child_process` instead of `worker_threads`.
According to our benchmarks, implementation using workers works ~ twice as fast

Reply:
API stability, there might be behavioural differences. Upgrading my app from node 10 to 11 (or whenever they remove the flag) shouldn't change the underlying implementation IMO. The docs for the module says it uses multiple processes, and just silently not doing that, depending on node version, seems risky to me

FWIW I could get behind setting it to `true` by default so it's possible to opt-out.

Reply:
> Upgrading my app from node 10 to 11 (or whenever they remove the flag) shouldn't change the underlying implementation IMO

I think that underlying implementation is not something that should concern an average library user. Public interface stays completely the same (which is crucial for all library users), while "engine" is getting faster. Same happens with the language and other libraries like React (Fiber rewrite) etc.

> The docs for the module says it uses multiple processes, and just silently not doing that, depending on node version, seems risky to me

I'm completely on the same page, but we're going to cover all that by specs so there should be no issues with unexpected regression bugs. 

I think @mjesun can give you more insights.

Reply:
Let's re-write the docs as "we'll choose the most-performant implementation that is possible". People can enforce one or another by explicitly passing the corresponding worker pool.

Reply:
Sounds good to me!